### ISSUE TITLE
Warnings with Poetry 1.2.0

### ISSUE THREAD

**Issue by ottobackwards** [cjolowicz/nox-poetry#873]
nox > Running session lint-3.8
nox > Creating virtual environment (virtualenv) using python3.8 in .nox/lint-3-8
nox > poetry export --format=requirements.txt --dev --without-hashes
The --dev option is deprecated, use the --with dev notation instead.

**Comment by edgarrmondragon** [comment]
Yeah, `dev` is a dependency group in `1.2`.

Related: #663

**Comment by johnthagen** [comment]
See:

- https://python-poetry.org/blog/announcing-poetry-1.2.0/#dependency-groups

**Comment by elbakramer** [comment]
I think warning happens due to this line 85:
https://github.com/cjolowicz/nox-poetry/blob/c79617548e67b57508543a92d22f7b80c5bf2d23/src/nox_poetry/poetry.py#L81-L91

Can it be replaced to this?:

```python
output = self.session.run_always( 
     "poetry", 
     "export", 
     "--format=requirements.txt", 
     "--with", "dev", 
     *[f"--extras={extra}" for extra in self.config.extras], 
     "--without-hashes", 
     external=True, 
     silent=True, 
     stderr=None, 
 ) 
```

**Comment by guysalt** [comment]
some one know whats going on with this issue?

i see a lot of PRs and Issues on supporting poetry 1.2, and i don't understand where it is blocked?

i can open a PR to fix this warning line but it seems like it is on purpose...

thanks

**Comment by johnthagen** [comment]
@guysalt This is fixed in 

- #1080

Current status is awaiting a merge/feedback from @cjolowicz, who has been busy but has recently merged in some other improvements

- https://github.com/cjolowicz/nox-poetry/pull/1080#issuecomment-1519032865

There is also some history about @cjolowicz thoughts about the future of `nox-poetry` all together referenced here. It's possible he may decide to pass the project off or add more maintainers to help move it forward.

- https://github.com/cjolowicz/nox-poetry/pull/1080#issuecomment-1518229068

**Comment by guysalt** [comment]
thank you so much.

hope it will be fixed soon.

**Comment by cjolowicz** [comment]
This was fixed in [nox-poetry 1.0.3](https://github.com/cjolowicz/nox-poetry/releases/tag/v1.0.3).
